compositor_api/src/types/from_color.rs

```diff
@@ -0,0 +1,268 @@
+use compositor_render::scene;
+
+use super::util::*;
+
+impl TryFrom<RGBAColor> for scene::RGBAColor {
+    type Error = TypeError;
+    fn try_from(value: RGBAColor) -> std::result::Result<Self, Self::Error> {
+        let s = &value.0.trim();
+        match s.chars().next() {
```

Comment:
Something like this would be more readable
```
if let Some(color) = parse_named_color() {
  return Ok(color)
}
if s.starts_with('#') {
  ...
  return
}
if s.starts_with('rgb(') {
  ...
  return
}
if s.starts_with('rgba(') {
  ...
  return
}
```